pip/req/req_set.py

```diff
@@ -123,6 +133,68 @@ def prep_for_dist(self):
         self.req_to_install.assert_source_matches_version()
 
 
+class SetupCfgDistribution(pkg_resources.Distribution):
+
+    def __init__(self, cfg, location=None):
+        self._cfg = cfg
+        project_name = cfg.get('metadata', 'name')
+        super(SetupCfgDistribution, self).__init__(
+            location=location, project_name=project_name)
+
+    @property
+    def _dep_map(self):
+        try:
+            return self.__dep_map
+        except AttributeError:
+            # requires
+            try:
+                requires = self._cfg.get('metadata', 'requires-dist')
+            except configparser.NoOptionError:
```

Comment:
Not caching was a bug, fixed. I think the function is small enough that breaking it up into little pieces will harm understandability rather than aid it.